### ISSUE TITLE
Color segmentation masks in Stardist_2D 

### ISSUE THREAD

**Issue by iRavkin** [HenriquesLab/ZeroCostDL4Mic#8]
I have completed all steps in Stardist_2D notebook and got correct segmentation results on the two images in "Stardist/Test - Images". The results include two images in "Stardist/Test - Masks" with the same names. These images are four-plane TIFs, where the first three planes encode the mask color. 

Then I repeated the whole process, but this time I added two of my own images in "Stardist/Test - Images". They were 16-bit and 8-bit versions of the same image of DAPI-stained nuclei. The resulting segmentation masks in "Stardist/Test - Masks" were the same for the original images, but for my images they had identical values in the first three planes. So, the segmentation image looked gray and the touching masks most often could not be distinguished.
![original](https://user-images.githubusercontent.com/55162514/78322314-39aa7900-7523-11ea-8edc-ddc259c86a97.png)
![masks](https://user-images.githubusercontent.com/55162514/78322324-3fa05a00-7523-11ea-86e0-649df7eb19ee.png)

Please, advise.

**Comment by guijacquemet** [comment]
Hi, 
Thanks you for reaching out. I am not quite sure I understand some of the things you are describing.
Are you doing the prediction on single images ? or on stacks of images ?

What do you mean by: "These images are four-plane TIFs, where the first three planes encode the mask color"

What do you mean by : "my images they had identical values in the first three planes" ?

**Comment by guijacquemet** [comment]
Also, stardist prediction does not generate color mask but instead images where all the identified ROI have a different gray value . In the result folder you can also find a .ZIP file that contains all the ROI. This file can be openned in imageJ / Fiji for instance.

**Comment by iRavkin** [comment]
Thank you for quick response. I am doing prediction on single images. Below are the original and predicted images for one of the two included test images. Apparently prediction generates a color mask, which would be more convenient for me to use than ROIs. This color mask is 1024x1024x4, where planes 0,1, and 2 correspond to RGB color seen in the second image. In the prediction image that was generated from my test image these planes were identical (R=G=B).
![cell migration R1 - Position 58_XY1562686154_Z0_T00_C1-image86-original](https://user-images.githubusercontent.com/55162514/78401013-a9187b00-75ac-11ea-8be6-eee8d2e450bc.png)
![cell migration R1 - Position 58_XY1562686154_Z0_T00_C1-image86-masks](https://user-images.githubusercontent.com/55162514/78401024-afa6f280-75ac-11ea-82e9-a30e6820f989.png)

**Comment by iRavkin** [comment]
I realize now that the color mask is not a predicted image, but the ground truth image. If it is present in the "Stardist/Test - Masks" folder it is not overwritten, but if it is not present, then prediction generates a 1024x1024x32bit image. When I read it with pyplot.imread it shows as uint8 (1024,1024,4) and the first three planes are identical. So, it can't encode more than 255 ROIs.
![cell migration R1 - Position 58_XY1562686154_Z0_T00_C1-image86-masks](https://user-images.githubusercontent.com/55162514/78414787-a761ae00-75d2-11ea-914f-d240cd606b17.png)
Please, let me know if I am doing something incorrectly.

**Comment by guijacquemet** [comment]
Thanks for the details. 
The images should be saved as single-channel 32-bit integer TIFF image. 
Could it be an issue with "pyplot.imread" ?   https://stackoverflow.com/questions/41051278/how-to-read-a-32-bit-integer-image-using-matplotlib

Can you open the generated images in Fiji ?

**Comment by iRavkin** [comment]
Yes, that was it. Both imread from skimage.io and open from PIL.Image read 32-bit tiff files correctly.
Thank you very much for help.

**Comment by guijacquemet** [comment]
Excellent, glad it works now.
To get good results with your own images do not forget to create you own training set so that the network you will train is optimized for your own data.
